build/build-plugins.js

```diff
@@ -23,158 +25,113 @@ const plugins = [
-  Toast: path.resolve(__dirname, '../js/src/toast.js'),
-  Tooltip: path.resolve(__dirname, '../js/src/tooltip.js')
-}
 
-const defaultPluginConfig = {
-  external: [
-    bsPlugins.Data,
-    bsPlugins.Base,
-    bsPlugins.EventHandler,
-    bsPlugins.SelectorEngine
-  ],
-  globals: {
-    [bsPlugins.Data]: 'Data',
-    [bsPlugins.Base]: 'Base',
-    [bsPlugins.EventHandler]: 'EventHandler',
-    [bsPlugins.SelectorEngine]: 'SelectorEngine'
-  }
+const utilPlugins = {
+  BackDrop: 'util/backdrop.js',
+  ScrollBarHelper: 'util/scrollbar.js',
+  sanitizeHtml: 'util/sanitizer.js',
+  Util: 'util/index.js'
 }
 
-const getConfigByPluginKey = pluginKey => {
-  switch (pluginKey) {
-    case 'Alert':
-    case 'Offcanvas':
-    case 'Tab':
-      return defaultPluginConfig
-
-    case 'Base':
-    case 'Button':
-    case 'Carousel':
-    case 'Collapse':
-    case 'Modal':
-    case 'ScrollSpy': {
-      const config = Object.assign(defaultPluginConfig)
-      config.external.push(bsPlugins.Manipulator)
-      config.globals[bsPlugins.Manipulator] = 'Manipulator'
-      return config
-    }
-
-    case 'Dropdown':
-    case 'Tooltip': {
-      const config = Object.assign(defaultPluginConfig)
-      config.external.push(bsPlugins.Manipulator, '@popperjs/core')
-      config.globals[bsPlugins.Manipulator] = 'Manipulator'
-      config.globals['@popperjs/core'] = 'Popper'
-      return config
-    }
-
-    case 'Popover':
-      return {
-        external: [
-          bsPlugins.Data,
-          bsPlugins.SelectorEngine,
-          bsPlugins.Tooltip
-        ],
-        globals: {
-          [bsPlugins.Data]: 'Data',
-          [bsPlugins.SelectorEngine]: 'SelectorEngine',
-          [bsPlugins.Tooltip]: 'Tooltip'
-        }
-      }
+const domPlugins = {
+  Data: 'dom/data.js',
+  EventHandler: 'dom/event-handler.js',
+  Manipulator: 'dom/manipulator.js',
+  SelectorEngine: 'dom/selector-engine.js'
+}
 
-    case 'Toast':
-      return {
-        external: [
-          bsPlugins.Data,
-          bsPlugins.Base,
-          bsPlugins.EventHandler,
-          bsPlugins.Manipulator
-        ],
-        globals: {
-          [bsPlugins.Data]: 'Data',
-          [bsPlugins.Base]: 'Base',
-          [bsPlugins.EventHandler]: 'EventHandler',
-          [bsPlugins.Manipulator]: 'Manipulator'
-        }
-      }
+const mainPlugins = {
+  BaseComponent: 'base-component.js',
+  Alert: 'alert.js',
+  Button: 'button.js',
+  Carousel: 'carousel.js',
+  Collapse: 'collapse.js',
+  Dropdown: 'dropdown.js',
+  Modal: 'modal.js',
+  Offcanvas: 'offcanvas.js',
+  Popover: 'popover.js',
+  ScrollSpy: 'scrollspy.js',
+  Tab: 'tab.js',
+  Toast: 'toast.js',
+  Tooltip: 'tooltip.js'
+}
 
-    default:
-      return {
-        external: []
-      }
-  }
+const nodePlugins = {
+  Popper: '@popperjs/core'
 }
 
-const utilObjects = new Set([
-  'Util',
-  'Sanitizer',
-  'Backdrop'
-])
+const bsPlugins = {
+  ...domPlugins,
+  ...utilPlugins,
```

Comment:
This will create files for utils (and mark them as external)? Is this intended because the old code wasn't doing this?

Reply:
Yes it is intended my my side. The old code was wrote 2 years ago, and hasn't been updated since then. However the codebase had faced drastic changes on imports, code separations etc, that were not mirrored in build-plugins.js 

We can discuss this, although I don't find something to worry about